File: adapter/exercise/ReportingService.java
package adapter.exercise;

public class ReportingService {
    private final ThirdPartyReportingService thirdPartyReportingService;

    public ReportingService(ThirdPartyReportingService thirdPartyReportingService) {
        this.thirdPartyReportingService = thirdPartyReportingService;
    }

    public CSVReport generateReport() {
        return thirdPartyReportingService.generateReport();
    }
}

File: typescript/adapter/exercises/reporting-service.ts
import { CSVReport } from "./cv-report";
import { ThirdPartyReportingService } from "./third-party-reporting-service";

export class ReportingService {
  private readonly thirdPartyReportingService: ThirdPartyReportingService;

  constructor(thirdPartyReportingService: ThirdPartyReportingService) {
    this.thirdPartyReportingService = thirdPartyReportingService;
  }

  generateReport(): CSVReport {
    return this.thirdPartyReportingService.generateReport();
  }
}
